### PR TITLE
Use createMock() and use import instead of FQCN

### PR DESCRIPTION
**Pull request by OskarStark**
| Q             | A
| ------------- | ---
| Branch?       | 4.4
| Bug fix?      | no
| New feature?  | no
| Deprecations? | no
| Tickets       | ---
| License       | MIT
| Doc PR        | ---

### CODE REVIEW

**Comment by jderusse** [comment]
For reference https://github.com/symfony/symfony/pull/20984

**Comment by OskarStark** [comment]
I checked all files manually on Github:
CleanShot 2021-01-22 at 14 02 24

Lets see if the tests pass

File: src/Symfony/Component/Security/Core/Tests/Authentication/Provider/DaoAuthenticationProviderTest.php [hunk 1]
```diff
@@ -51,13 +51,13 @@ public function testRetrieveUserWhenUsernameIsNotFound()
...
         $this->expectException(\Symfony\Component\Security\Core\Exception\AuthenticationServiceException::class);
-        $userProvider = $this->getMockBuilder(UserProviderInterface::class)->getMock();
+        $userProvider = $this->createMock(UserProviderInterface::class);
         $userProvider->expects($this->once())
                      ->method('loadUserByUsername')
                      ->willThrowException(new \RuntimeException())
         ;
 
-        $provider = new DaoAuthenticationProvider($userProvider, $this->getMockBuilder(\Symfony\Component\Security\Core\User\UserCheckerInterface::class)->getMock(), 'key', $this->getMockBuilder(\Symfony\Component\Security\Core\Encoder\EncoderFactoryInterface::class)->getMock());
+        $provider = new DaoAuthenticationProvider($userProvider, $this->createMock(\Symfony\Component\Security\Core\User\UserCheckerInterface::class)->getMock(), 'key', $this->getMockBuilder(\Symfony\Component\Security\Core\Encoder\EncoderFactoryInterface::class));
```
Comment (OskarStark) [inline src/Symfony/Component/Security/Core/Tests/Authentication/Provider/DaoAuthenticationProviderTest.php]:
```suggestion
        $provider = new DaoAuthenticationProvider($userProvider, $this->createMock(\Symfony\Component\Security\Core\User\UserCheckerInterface::class), 'key', $this->createMock(\Symfony\Component\Security\Core\Encoder\EncoderFactoryInterface::class));
```

File: src/Symfony/Bridge/Doctrine/Tests/DataCollector/DoctrineDataCollectorTest.php [hunk 2]
```diff
@@ -249,7 +249,7 @@ private function createCollector($queries)
             ->method('getConnection')
             ->willReturn($connection);
 
-        $logger = $this->getMockBuilder(\Doctrine\DBAL\Logging\DebugStack::class)->getMock();
+        $logger = $this->createMock(\Doctrine\DBAL\Logging\DebugStack::class);
```
Comment (derrabus) [inline src/Symfony/Bridge/Doctrine/Tests/DataCollector/DoctrineDataCollectorTest.php]: I feel urged to replace the FQCN with an import, but apparently you could resist. 👏🏻 😅

Comment (OskarStark) [reply src/Symfony/Bridge/Doctrine/Tests/DataCollector/DoctrineDataCollectorTest.php]: Will do, but the change right now is massive, lets check if tests are green and afterwards, I can replace the FQCN, if you prefer in this PR, or in a new one

Comment (derrabus) [reply src/Symfony/Bridge/Doctrine/Tests/DataCollector/DoctrineDataCollectorTest.php]: I think we should separate the steps. 🙈

Comment (OskarStark) [reply src/Symfony/Bridge/Doctrine/Tests/DataCollector/DoctrineDataCollectorTest.php]: I think not, let's make it clean 😜

**Comment by OskarStark** [comment]
I am not done yet 😬

File: src/Symfony/Bridge/Twig/Tests/Command/DebugCommandTest.php [hunk 3]
```diff
@@ -264,12 +265,12 @@ public function getDebugTemplateNameTestData()
...
- ----------- -------------------------------------- 
-  Namespace   Paths                                 
- ----------- -------------------------------------- 
+ ----------- --------------------------------------
+  Namespace   Paths
+ ----------- --------------------------------------
   @Twig       templates/bundles/TwigBundle%e%A
               vendors/twig-bundle/Resources/views%e%A
- ----------- -------------------------------------- 
+ ----------- --------------------------------------
```
Comment (nicolas-grekas) [inline src/Symfony/Bridge/Twig/Tests/Command/DebugCommandTest.php]: whitespace in this file should be kept

**Comment by nicolas-grekas** [comment]
This is going to be fun to merge into 5.2 (I'd way after the last 5.1 before merging this if I were you).

May I suggest ignoring all conflicts when doing the merge?

File: src/Symfony/Bridge/Twig/Tests/Command/DebugCommandTest.php (hunk 3 above)
Comment (OskarStark) [reply src/Symfony/Bridge/Twig/Tests/Command/DebugCommandTest.php]: Yep need to double check everything again before merge

**Comment by OskarStark** [comment]
Well let's merge this when 5.1 is EOL and merge it direct into 5.2? Will try to upmerge it beforehand and see how many conflicts we have.

File: src/Symfony/Bridge/Monolog/Tests/Handler/ConsoleHandlerTest.php [hunk 4]
```diff
@@ -131,14 +134,15 @@ public function testVerbosityChanged()
     public function testGetFormatter()
     {
         $handler = new ConsoleHandler();
-        $this->assertInstanceOf(\Symfony\Bridge\Monolog\Formatter\ConsoleFormatter::class, $handler->getFormatter(),
+        $this->assertInstanceOf(
+            ConsoleFormatter::class, $handler->getFormatter(),
```
Comment (OskarStark) [inline src/Symfony/Bridge/Monolog/Tests/Handler/ConsoleHandlerTest.php:138]:
```suggestion
        $this->assertInstanceOf(ConsoleFormatter::class, $handler->getFormatter(),
```

File: src/Symfony/Bridge/Twig/Tests/Command/DebugCommandTest.php (hunk 3 above)
Comment (OskarStark) [reply src/Symfony/Bridge/Twig/Tests/Command/DebugCommandTest.php]: Fixed

File: src/Symfony/Bundle/FrameworkBundle/Tests/Controller/ControllerTraitTest.php [hunk 5]
```diff
@@ -518,7 +533,7 @@ public function testCreateFormBuilder()
 
     public function testGetDoctrine()
     {
-        $doctrine = $this->getMockBuilder(ManagerRegistry::class)->getMock();
+        $doctrine = $this->createMock(ManagerRegistry::class);
```
Comment (OskarStark) [inline src/Symfony/Bundle/FrameworkBundle/Tests/Controller/ControllerTraitTest.php]: @nicolas-grekas @derrabus any idea why this test is failing with PHP 7.4? 🤔

Comment (jderusse) [reply src/Symfony/Bundle/FrameworkBundle/Tests/Controller/ControllerTraitTest.php]: `createMock` calls `disallowMockingUnknownTypes` on the mockBuilder?

Comment (OskarStark) [reply src/Symfony/Bundle/FrameworkBundle/Tests/Controller/ControllerTraitTest.php]: Yes, but shouldn't we fix this somehow or mark it as skipped?

Comment (xabbuh) [reply src/Symfony/Bundle/FrameworkBundle/Tests/Controller/ControllerTraitTest.php]: see #39968

**Comment by OskarStark** [comment]
Fabbot failure false positive.

@nicolas-grekas feel free to merge this in 4.4 only, I will take care of a new PR against 5.2 then by handling the conflicts, fine?

**Comment by nicolas-grekas** [comment]
Thank you @OskarStark 
Now merged up to 5.x
I'd appreciate if you could help me fix tests before doing this work on 5.1+ :)

**Comment by OskarStark** [comment]
Sure will have a look later 👍🏻

**Pull request merged**